code/modules/mob/living/simple_animal/hostile/mining_mobs/goliath.dm

```diff
@@ -10,7 +10,7 @@
 	icon_gib = "syndicate_gib"
 	mob_biotypes = MOB_ORGANIC|MOB_BEAST
 	mouse_opacity = MOUSE_OPACITY_ICON
-	move_to_delay = 40
+	move_to_delay = 20
```

Comment:
This isn't the proper fix here. You're resolving the symptom, rather then the cause, which just kinda papers over the core issue if that makes sense. (Also their initial move speed was 40 deciseconds, I just fucked up bigtime when doing some reasoning about byond procs)

That said, I do actually know what's wrong here. If a loop is queued while another copy of it is still in existence, it will just override the old loop, and reset its timer. Should look at https://github.com/BeeStation/BeeStation-Hornet/pull/6845, might be worth taking some ideas from there